protobuf/echo/Echo.hpp

```diff
@@ -10,20 +10,46 @@
 #include "protobuf/echo/EchoErrorPolicy.hpp"
 #include "protobuf/echo/Proto.hpp"
 #include "protobuf/echo/Serialization.hpp"
+#include <cstdint>
 
 namespace services
 {
     class Echo;
+    class ServiceId;
     class Service;
     class ServiceProxy;
 
-    class Service
-        : public infra::Observer<Service, Echo>
+    class ServiceIdAccess
+    {
+    public:
+        static uint32_t GetId(const ServiceId& serviceId);
+    };
```

Comment:
If anyone can use ServiceIdAccess, then you might as well just add GetId() to ServiceId instead of adding this class